docs/user-guide/synthetic.rst

```diff
@@ -0,0 +1,113 @@
+        PacketField(name="DATAU", data_type="uint", bit_length=32),
+        PacketField(name="DATAI", data_type="int", bit_length=16),
+        PacketField(name="DATAF", data_type="float", bit_length=64),
+        PacketField(name="DATAS", data_type="str", bit_length=8)
+        PacketField(name="DATAFLAG", data_type='uint', bit_length=1),
+        PacketArray(
+                name="SENSOR_GRID",
+                data_type="uint",
+                bit_length=16,
+                array_shape=(32, 32),
+                array_order="C",
+            )
+   ])
+
+Once you've done that you'll have to define a dictionary whose keys are the packet field names and whose values are your data arrays.
+
+.. code-block:: python
+
+    num_packets = 1000
+    datau = np.arange(0, num_packets, dtype=np.uint32)
+    datai = np.arange(1, num_packets + 1, dtype=np.int16)
+    dataf = np.arange(2, num_packets + 2, dtype=np.float64)
+    datas = np.arange(97, num_packets + 97, dtype=np.uint8)
+    dataflag = np.zeros(num_packets, dtype=np.bool)
+    sensor_grid = np.arange(0, num_packets * 32 * 32, dtype=np.uint16).reshape(num_packets, 32, 32)
+    data = {
+        "DATAU": datau,
+        "DATAI": datai,
+        "DATAF": dataf,
+        "DATAS": datas,
+        "DATAFLAG": dataflag,
+        "SENSOR_GRID": sensor_grid
+    }
+
+If you have a non-standard bit_length (e.g. 6, 14), simply select a dtype that will fit the data (e.g. 8, 16, respectively).
+Once you have this data dictionary defined you can write to a file
+
+.. code-block:: python
+
+    pkt_type = 0  # For telemetry (or reporting), set to ‘0’
+    apid = 0x084  # the Application process identifier
+    sec_header_flag = 0  # identicates the presence or absence of a secondary header 
+    seq_flag = 0  # Set to 1 if continuation of a packet, 0 otherwise
+    pkt.to_file("test.bin", pkt_type, apid, sec_header_flag, seq_flag, data)
+
+For a description of the CCSDS header see :ref:`ccsds_standard`.
+You can parse the file using the same object if you'd like to check its contents with::
+
+    data = pkt.load("test.bin")
+
+Variable Length
+===============
+Variable length packets are packets which may have a different length each time.
+They include at least one variable length field defined by `~ccsdspy.PacketArray` which can either set `array_shape="expand"` (causing the field to grow to fill the packet) or
+`array_shape="other_field"` (causes the field named `other_field` to set the number of elements in this array).
+
+.. code-block:: python
+
+    pkt = VariableLength(
+            [
+                PacketField(name="DATAU", data_type="uint", bit_length=32),
+                PacketArray(
+                    name="VARARRAY",
+                    data_type="uint",
+                    bit_length=8,
+                    array_shape="expand",
+                ),
+            ]
+        )
+
+Since the array is a variable length we cannot use a `~numpy.ndarray` so we will use a list.
```

Comment:
You can use an ndarray of `dtype=object` for arrays of this type; this is what we return for variable length arrays. Can you adjust this sentence to say:
```
Since the array is a variable length we cannot use a typical 
`~numpy.ndarray`. Instead, we can use either a `list` or a 
`~numpy.ndarray` with `dtype=object` referencing other arrays
at each element. In this example, we will use a list.

```